packages/backend/src/open_payments/quote/service.ts

```diff
@@ -0,0 +1,365 @@
+      }
+    } else if (
+      !options.receiveAmount ||
+      options.receiveAmount.value <= BigInt(0)
+    ) {
+      return QuoteError.InvalidAmount
+    }
+  } else {
+    return QuoteError.InvalidDestination
+  }
+
+  const account = await deps.accountService.get(options.accountId)
+  if (!account) {
+    return QuoteError.UnknownAccount
+  }
+  if (options.sendAmount) {
+    if (
+      options.sendAmount.assetCode !== account.asset.code ||
+      options.sendAmount.assetScale !== account.asset.scale
+    ) {
+      return QuoteError.InvalidAmount
+    }
+  }
+
+  const plugin = deps.makeIlpPlugin({
+    sourceAccount: account,
+    unfulfillable: true
+  })
+
+  try {
+    await plugin.connect()
+    const setupOptions: Pay.SetupOptions = { plugin }
+    if (options.receivingPayment) {
+      setupOptions.destinationPayment = options.receivingPayment
+    } else {
+      setupOptions.destinationAccount = options.receivingAccount
+      if (options.receiveAmount) {
+        setupOptions.amountToDeliver = {
+          value: options.receiveAmount.value,
+          assetCode: options.receiveAmount.assetCode,
+          assetScale: options.receiveAmount.assetScale
+        }
+      }
+    }
+    let destination: Pay.ResolvedPayment
+    try {
+      destination = await Pay.setupPayment(setupOptions)
+    } catch (err) {
+      if (err === Pay.PaymentError.QueryFailed) {
+        return QuoteError.InvalidDestination
+      }
+      throw err
+    }
+    if (!destination.destinationPaymentDetails) {
+      deps.logger.warn(
+        {
+          options
+        },
+        'missing incoming payment'
+      )
+      throw new Error('missing incoming payment')
+    }
+
+    if (options.receiveAmount) {
+      if (
+        options.receiveAmount.assetScale !==
+          destination.destinationAsset.scale ||
+        options.receiveAmount.assetCode !== destination.destinationAsset.code
+      ) {
+        return QuoteError.InvalidAmount
+      }
+    }
+
+    const prices = await deps.ratesService.prices().catch((_err: Error) => {
+      throw new Error('missing prices')
+    })
+    const quoteOptions: Pay.QuoteOptions = {
+      plugin,
+      destination,
+      sourceAsset: {
+        scale: account.asset.scale,
+        code: account.asset.code
+      },
+      slippage: deps.slippage,
+      prices
+    }
+    assert.ok(quoteOptions.destination.destinationPaymentDetails)
+    const receivingPaymentAmount = destination.destinationPaymentDetails
+      .incomingAmount
+      ? destination.destinationPaymentDetails.incomingAmount.value -
+        destination.destinationPaymentDetails.receivedAmount.value
+      : undefined
+    if (options.sendAmount) {
+      quoteOptions.amountToSend = options.sendAmount.value
+      quoteOptions.destination.destinationPaymentDetails.incomingAmount = undefined
+    } else if (options.receiveAmount) {
+      if (options.receivingPayment) {
+        quoteOptions.amountToDeliver = options.receiveAmount.value
```

Comment:
Why is this not outside that `if` statement, i.e. above line 177?

Reply:
https://github.com/interledgerjs/interledgerjs/tree/master/packages/pay#quoteoptions
>Either `amountToSend`, `amountToDeliver`, or `destinationPayment` must be provided, in order to determine how much to pay.

For `receivingAccount` + `receiveAmount` quotes, the `destinationPayment` was created with the `receiveAmount` as the `incomingAmount`, which the `assert` in the `else` is checking.